src/test/java/uk/ac/cam/cl/dtg/isaac/api/UsersFacadeIT.java

```diff
@@ -0,0 +1,136 @@
+        replay(mockRequest);
+        HttpServletResponse mockResponse = niceMock(HttpServletResponse.class);
+
+        String userObjectString = String.format("{\"registeredUser\":{\"loggedIn\":true,\"email\":\"%1$s\",\"dateOfBirth\":\"%3$s\",\"password\":\"%2$s\",\"familyName\":\"%4$s\",\"givenName\":\"%5$s\"},\"userPreferences\":{},\"passwordCurrent\":null}",
+                email, password, dob, familyName, givenName);
+
+        Response response = null;
+        try {
+            response = usersFacade.createOrUpdateUserSettings(mockRequest, mockResponse, userObjectString);
+        } catch (InvalidKeySpecException e) {
+            fail("Unhandled InvalidKeySpecException during test");
+        } catch (NoSuchAlgorithmException e) {
+            fail("Unhandled NoSuchAlgorithmException during test");
+        }
+        assertEquals(Response.Status.BAD_REQUEST.getStatusCode(), response.getStatus());
+    }
+
+    private static Stream<Arguments> invalidRegistrationParameters() {
+        // Email, Password, DOB, familyName, givenName
+        return Stream.of(
+//                Arguments.of("new-student@test.com", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
+                // Isaac addresses are forbidden
+                Arguments.of("new-student@isaaccomputerscience.org", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
+                // Email field cannot be empty
+                Arguments.of("", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
+                // Email field cannot have conecutive .s
+                Arguments.of("new-student@test..com", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
+                // Email field must contain an @
+                Arguments.of("new-student.test.com", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
+                // Email field must match an @x.y pattern
+                Arguments.of("new-student@testcom", "password", "2000-01-01T00:00:00.000Z", "Test", "Test"),
+                // Family name cannot be empty
+                Arguments.of("new-student@test.com", "password", "2000-01-01T00:00:00.000Z", "", "Test"),
+                // Family name cannot exceed maximum length
+                Arguments.of("new-student@test.com", "password", "2000-01-01T00:00:00.000Z", "01abcdefghijklmn02abcdefghijklmn03abcdefghijklmn04abcdefghijklmn05abcdefghijklmn06abcdefghijklmn07abcdefghijklmn08abcdefghijklmn09abcdefghijklmn10abcdefghijklmn11abcdefghijklmn12abcdefghijklmn13abcdefghijklmn14abcdefghijklmn15abcdefghijklmn16abcdefghijklmn", "Test"),
+                // Family name cannot contain forbidden characters
+                Arguments.of("new-student@test.com", "password", "2000-01-01T00:00:00.000Z", "Te*st", "Test"),
+                // Given name cannot be empty
+                Arguments.of("new-student@test.com", "password", "2000-01-01T00:00:00.000Z", "Test", ""),
```

Comment:
It seems like the rules for the first name are the same as the ones for the second name. It may be worth having a data provider for invalid names, and then two tests that use the same provider, one of them to test invalid first names and another one to test invalid last names. That way it's clearer and it avoids repetition.